scripts/test_bridge_detection.py

```
@@ -0,0 +1,421 @@
+            print(f"\n🎯 Hypothesis Validation:")
+            print(f"  Positive cases (should be bridges): {positive_bridges}/{len(positive_cases)}")
+            print(f"  Negative cases (should NOT be bridges): {negative_bridges}/{len(negative_cases)}")
+            
+            if positive_bridges > 0 and negative_bridges == 0:
+                print(f"  ✅ HYPOTHESIS VALIDATED: Bridge detection working correctly!")
+            else:
+                print(f"  ⚠️  Hypothesis needs refinement")
+        
+    except Exception as e:
+        logger.error(f"❌ Bridge detection test failed: {e}")
+        raise click.ClickException(str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)